fairlearn/metrics/_make_derived_metric.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import functools
+import inspect
+import numpy as np
+from typing import Callable, List, Union
+
+from ._metric_frame import MetricFrame
+
+transform_options = [
+    'difference',
+    'group_min',
+    'group_max',
+    'ratio',
+]
+
+_METRIC_CALLABLE_ERROR = "Supplied metric object must be callable"
+_METHOD_ARG_ERROR = "Callables which accept a 'method' argument " \
+    "may not be passed to make_derived_metric()"
+_INVALID_TRANSFORM = "Transform must be one of {0}".format(transform_options)
+
+
+class _DerivedMetric:
+    def __init__(self,
+                 *,
+                 metric: Callable[..., Union[float, int]],
+                 transform: str,
+                 sample_param_names: List[str]):
+
+        if not callable(metric):
+            raise ValueError(_METRIC_CALLABLE_ERROR)
```

Comment:
Is `ValueError` appropriate here (and for the next check on the signature)?

Reply:
LGTM